src/tune-hyperparameters/src/main/scala/HyperparamBuilder.scala

```diff
@@ -54,6 +54,36 @@ class DoubleRangeHyperParam(min: Double, max: Double, seed: Long = 0)
 
 }
 
+object HyperParamUtils {
+  /**
+    * Returns a range hyper param by matching to the given input type.
+    * @param min The min value of the range
+    * @param max The max value of the range
+    * @param seed The random number seed.
+    * @return A RangeHyperParam matched to the given type for min and max values.
+    */
+  def getRangeHyperParam(min: Any, max: Any, seed: Long = 0): RangeHyperParam[_] = {
+    min match {
```

Comment:
You can use scala pattern matching to your advantage here: try something like

(min, max) match {
    case (a: Float, b: Float) => floatRange(a,b)
   ......

Reply:
good idea, done!